ui/packages/atlasmap/src/Atlasmap/utils/field.ts

```diff
@@ -85,100 +85,126 @@ export function getConstantTypeIndex(constVal: string): number {
-  cfg.propertyDoc.removeField(field);
+  if (isSource) {
+    cfg.sourcePropertyDoc.removeField(field);
+  } else {
+    cfg.targetPropertyDoc.removeField(field);
+  }
   cfg.mappingService.notifyMappingUpdated();
 }
 
 export function editProperty(
   propName: string,
-  propValue: string,
   propType: string,
   propScope: string,
   newName: string,
+  isSource: boolean,
 ): void {
   const cfg = ConfigModel.getConfig();
-  let field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName,
-  );
+  let field = isSource
+    ? cfg.sourcePropertyDoc.getField(
+        cfg.sourcePropertyDoc.pathSeparator + propName,
+      )
+    : cfg.targetPropertyDoc.getField(
+        cfg.targetPropertyDoc.pathSeparator + propName,
+      );
   if (!field) {
     return;
   }
   if (propName !== newName) {
     field.name = newName;
   }
-  field.value = propValue;
   field.type = propType;
   field.scope = propScope;
-  cfg.propertyDoc.updateField(field, "");
-  cfg.mappingService.notifyMappingUpdated();
-}
-
-export function getPropertyValue(propName: string): string {
-  const cfg = ConfigModel.getConfig();
-  const field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName.split(" ")[0],
-  );
-  if (!field) {
-    return "";
+  if (isSource) {
+    cfg.sourcePropertyDoc.updateField(field, "");
+  } else {
+    cfg.targetPropertyDoc.updateField(field, "");
   }
-  return field.value;
+  cfg.mappingService.notifyMappingUpdated();
 }
 
-export function getPropertyType(propName: string): string {
+export function getPropertyType(propName: string, isSource: boolean): string {
   const cfg = ConfigModel.getConfig();
-  const field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName,
-  );
+  const field = isSource
+    ? cfg.sourcePropertyDoc.getField(
+        cfg.sourcePropertyDoc.pathSeparator + propName,
+      )
+    : cfg.targetPropertyDoc.getField(
+        cfg.targetPropertyDoc.pathSeparator + propName,
+      );
   if (!field) {
     return "";
   }
   return field.type;
 }
 
-export function getPropertyScope(propName: string): string {
+export function getPropertyScope(propName: string, isSource: boolean): string {
   const cfg = ConfigModel.getConfig();
-  const field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName,
-  );
+  const field = isSource
+    ? cfg.sourcePropertyDoc.getField(
+        cfg.sourcePropertyDoc.pathSeparator + propName,
+      )
+    : cfg.targetPropertyDoc.getField(
+        cfg.targetPropertyDoc.pathSeparator + propName,
+      );
   if (!field) {
     return "";
   }
   return field.scope;
 }
 
-export function getPropertyTypeIndex(propName: string): number {
+export function getPropertyTypeIndex(
+  propName: string,
+  isSource: boolean,
+): number {
   const cfg = ConfigModel.getConfig();
-  const field = cfg.propertyDoc.getField(
-    cfg.propertyDoc.pathSeparator + propName,
-  );
+  const field = isSource
```

Comment:
'cause yeah, I scrolled down from the last comment and it's 8 checks along these same lines.  Maybe there's a nicer way?